Spyder fails to launch kernel for pixi environments when `PIXI_HOME` is set

Possibly related: https://github.com/spyder-ide/spyder/issues/25564

About PIXI_HOME
---

https://pixi.prefix.dev/latest/reference/environment_variables/#configurable-environment-variables shows that `PIXI_HOME` is supposed to be a directory not an executable. And the executable goes inside `PIXI_HOME/bin` when you set the variable and do the `curl ... | sh` to install `pixi`.

So, I have set PIXI_HOME on my system to a folder in my D: drive to avoid the default pixi install location in $HOME in C: drive.

Problem with `find_pixi`
---

The `find_pixi` function looks at the `PIXI_HOME` but then returns that directly instead of doing `os.path.join(pixi, "bin", "pixi.exe")`:
https://github.com/spyder-ide/spyder/blob/2a471f5565b10f2ce20c54984f58d2e4b49706c8/spyder/utils/conda.py#L114-L121

Incorrect error message
---
Because of the problem in `find_pixi`, spyder fails to create ipython console for pixi environments on my machine, and fails with the incorrect message about `$APPDATA/jupyter/runtime` being readonly (it is not, I am able to create files there).:

<img width="1579" height="238" alt="Image" src="https://github.com/user-attachments/assets/bef90d21-5e56-4d07-a88f-f9ebc4425321" />

---

Looking at the minimal logs (`spyder --debug-info minimal`), we see the actual problem that the access is denied is probably because it is trying to run a folder as a subprocess (I can paste more of this if needed but I think the part I have pasted below is the problem).

```
2026-04-11 02:01:11,964 [INFO] [spyder.plugins.ipythonconsole.utils.kernelspec] -> Kernel command: ['D:\\Prog\\python\\pixi', 'run', '--environment', 'default', '--manifest-path', 'D:\\Prog\\python\\envs\\py\\pixi.toml', 'D:\\Prog\\python\\envs\\py\\.pixi\\envs\\default\\python.exe', '-Xfrozen_modules=off', '-m', 'spyder_kernels.console', '-f', '{connection_file}']
2026-04-11 02:01:11,966 [ERROR] [traitlets] -> Failed to run command:
['D:\\Prog\\python\\pixi', 'run', '--environment', 'default', '--manifest-path', 'D:\\Prog\\python\\envs\\py\\pixi.toml', 'D:\\Prog\\python\\envs\\py\\.pixi\\envs\\default\\python.exe', '-Xfrozen_modules=off', '-m', 'spyder_kernels.console', '-f', 'C:\\Users\\saaket\\AppData\\Roaming\\jupyter\\runtime\\kernel-d2923c2a52b9.json']
    PATH='D:\\Prog\\python\\pixi\\envs\\spyder\\Library\\bin;D:\\Prog\\python\\pixi\\envs\\spyder;D:\\Prog\\python\\pixi\\envs\\spyder\\Library/mingw-w64/bin;D:\\Prog\\python\\pixi\\envs\\spyder\\Library/usr/bin;D:\\Prog\\python\\pixi\\envs\\spyder\\Library/bin;D:\\Prog\\python\\pixi\\envs\\spyder\\Scripts;D:\\Prog\\python\\pixi\\envs\\spyder\\bin;D:\\Prog\\python\\envs\\py\\.pixi\\envs\\default;D:\\Prog\\python\\envs\\py\\.pixi\\envs\\default\\Library/mingw-w64/bin;D:\\Prog\\python\\envs\\py\\.pixi\\envs\\default\\Library/usr/bin;D:\\Prog\\python\\envs\\py\\.pixi\\envs\\default\\Library/bin;D:\\Prog\\python\\envs\\py\\.pixi\\envs\\default\\Scripts;D:\\Prog\\python\\envs\\py\\.pixi\\envs\\default\\bin;C:\\Program Files\\PowerShell\\7;C:\\windows\\system32;C:\\windows;C:\\windows\\System32\\Wbem;C:\\windows\\System32\\WindowsPowerShell\\v1.0\\;C:\\windows\\System32\\OpenSSH\\;C:\\Program Files\\dotnet\\;D:\\Windows Kits\\10\\Windows Performance Toolkit\\;C:\\Program Files\\Git\\cmd;C:\\Program Files\\PowerShell\\7\\;C:\\Program Files\\Docker\\Docker\\resources\\bin;C:\\Program Files\\nodejs\\;D:\\msys64\\ucrt64\\bin;D:\\msys64\\usr\\bin;D:\\Prog\\xdg\\bin;D:\\Prog\\python\\pixi\\bin;D:\\Prog\\rust\\cargo\\bin;D:\\Soft\\mpv;C:\\Users\\saaket\\AppData\\Local\\Programs\\Microsoft VS Code\\bin;C:\\Users\\saaket\\AppData\\Local\\Microsoft\\WindowsApps;C:\\Users\\saaket\\.dotnet\\tools;C:\\Users\\saaket\\AppData\\Roaming\\npm;D:\\msys64\\ucrt64\\bin;D:\\msys64\\usr\\bin;D:\\Prog\\xdg\\bin;D:\\Prog\\python\\pixi\\bin;D:\\Prog\\rust\\cargo\\bin;D:\\Soft\\mpv;C:\\Users\\saaket\\AppData\\Local\\Programs\\Microsoft VS Code\\bin;C:\\Users\\saaket\\AppData\\Local\\Microsoft\\WindowsApps;C:\\Users\\saaket\\.dotnet\\tools;C:\\Users\\saaket\\AppData\\Roaming\\npm'
    with kwargs:
{'stdin': -1, 'stdout': -1, 'stderr': -1, 'cwd': None, 'close_fds': False}

2026-04-11 02:01:11,966 [ERROR] [traitlets] -> [WinError 5] Access is denied
```

Potential fix
---

If I edit `spyder.utils.conda` file and change `find_pixi` to the following, the ipython console seems to load fine
```python
def find_pixi(pyexec=None):
    """
    Find pixi executable.

    `pyexec` is a python executable, the relative location from which to
    attempt to locate a pixi executable.
    """
    pixi_home = os.environ.get('PIXI_HOME', None)
    pixi_exec = 'pixi.exe' if WINDOWS else 'pixi'

    if pixi_home is None:
        pixi = find_program(pixi_exec)
    else:
        pixi = os.path.join(pixi_home, "bin", pixi_exec)

    return pixi
```

I can submit a PR with this change if this looks okay.

--- a/spyder/utils/conda.py
+++ b/spyder/utils/conda.py
@@ -110,13 +110,13 @@ def find_pixi(pyexec=None):
     `pyexec` is a python executable, the relative location from which to
     attempt to locate a pixi executable.
     """
-    # Try the `PIXI_HOME` environment variable
-    pixi = os.environ.get('PIXI_HOME', None)
+    pixi_home = os.environ.get('PIXI_HOME', None)
+    pixi_exec = 'pixi.exe' if WINDOWS else 'pixi'
 
-    # Next try searching for the executable in default paths
-    if pixi is None:
-        pixi_exec = 'pixi.exe' if WINDOWS else 'pixi'
+    if pixi_home is None:
         pixi = find_program(pixi_exec)
+    else:
+        pixi = os.path.join(pixi_home, "bin", pixi_exec)
 
     return pixi
 

